cyipopt/scipy_interface.py

```diff
@@ -596,8 +609,11 @@ def minimize_ipopt(fun,
     # Rename some default scipy options
     replace_option(options, b'disp', b'print_level')
     replace_option(options, b'maxiter', b'max_iter')
-    if getattr(options, 'print_level', False) is True:
-        options[b'print_level'] = 1
+    if b'print_level' in options:
+        if options[b'print_level'] is True:
+            options[b'print_level'] = 1
+        elif options[b'print_level'] is False:
+            options[b'print_level'] = 0
```

Comment:
This change doesn't appear to make any difference to the way the code functions, except increasing the number of lines of code to do the same thing. Am I missing something? If so, please can you explain why this change is needed?

Reply:
Ipopt print levels are integers between 1 and 12, with the default being 5: https://coin-or.github.io/Ipopt/OPTIONS.html#OPT_print_level

Reply:
I think we should use something like the following. This will ensure that if a `print_level` of `True` is used that the default print level is used, not breaking previous code. It will also allow users to manually select the exact desired print level as per the Ipopt options.

```suggestion
    if getattr(options, 'print_level') is True:
        options[b'print_level'] = 5
    else:
        print_level = getattr(options, 'print_level' , 5)
        if not isinstance(print_level, int) or (print_level < 1) or (print_level > 12):
            raise ValueError(f'`print_level` option must be an int between 1 and 12, not {print_level}')
        options[b'print_level'] = print_level
            
```

Reply:
I thought it this way: The original version would force the `print_level` to be one if the user specify this parameter, no matter what value is chosen. I tried to make it work like this: 
1. 'disp': True -> 'print_level': 1
2. 'disp': False -> 'print_level': 0
3. no 'disp' or 'print_level' -> 'print_level': 0 (according to Ipopt's document, a `print_level` of value `0` is acceptable)
4. 'disp'/'print_level': 0-12 -> 'print_level': 0-12

About the default `print_level` in the second case, I did have some doubts. I wondered before why Ipopt said that `print_level` was by default `5`, but that was not how it actually behavioured. So when I was changing these codes, I hesitated about whether to set the default value to `1` or `5`. I chose `1` eventually because it was the original setting.